File: c++/2467.cpp
#include <iostream>

using namespace std;

int arr[100000];

int main() {
  ios_base::sync_with_stdio(0);
  cin.tie(0);

  freopen("example.txt", "r", stdin);
  int N;
  cin >> N;

  for (int i = 0; i < N; i++) {
    cin >> arr[i];
  }

  int start = 0;
  int end = N - 1;
  int min = abs(arr[start] + arr[end]);
  int answer[2];
  answer[0] = arr[start];
  answer[1] = arr[end];

  while (start < end) {
    int sum = arr[start] + arr[end];
    int now = abs(sum);
    if (now <= min) {
      min = now;
      answer[0] = arr[start];
      answer[1] = arr[end];
    }

    if (sum < 0) {
      start++;
    } else {
      end--;
    }
  }

  cout << answer[0] << ' ' << answer[1];

  return 0;
}

File: js/2467.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
const arr = inputs[1].split(' ').map(Number);

if (arr[0] > 0) {
  console.log(arr[0], arr[1]);
} else if (arr[N - 1] < 0) {
  console.log(arr[N - 2], arr[N - 1]);
} else {
  let start = 0;
  let end = N - 1;
  let min = Math.abs(arr[start] + arr[end]);
  const answer = [arr[start], arr[end]];

  while (start < end) {
    if (Math.abs(arr[start] + arr[end]) <= min) {
      min = Math.abs(arr[start] + arr[end]);
      answer[0] = arr[start];
      answer[1] = arr[end];
    }

    if (Math.abs(arr[start]) < Math.abs(arr[end])) {
      end--;
    } else {
      start++;
    }
  }

  console.log(answer.join(' '));
}
